### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by Jyrno42**
- Use Yarn
- use node 8-11 in CI
- Update dependencies
- Add prettier
- Fix linting issues
- Fix examples
- Fix webpack/babel config

### CODE REVIEW

**Comment by simonschmidt** [comment]
:+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,34 +2,41 @@
...
 - Add react 16 to supported versions [issue #40, PR #51]
 - Removed BrowserModal indirection and merged BrowserModal w/ Modal [issue #35, PR #51]
 - Added prop `className` to `Modal` [issue #50, PR #51]
   - Added prop `modalClassName` to `Modal.Dialog`
 - Added prop `wrapperClassName` to `Modal`
-- Added node 9 to supported development engines
+- Update dependencies
+- Use yarn
+- Use node 8-12 in CI
+- Add prettier
```
Comment (simonschmidt) [inline CHANGELOG.md]: :+1:

File: examples/components/examples/Basic.js [hunk 2]
```diff
@@ -27,23 +26,28 @@ class BasicModalExample extends Component {
             e.preventDefault();
         }
 
-        this.setState({
-            isOpen: !this.state.isOpen
-        });
+        this.setState((prevState) => ({
```
Comment (simonschmidt) [inline examples/components/examples/Basic.js]: :+1:

**Comment by simonschmidt** [comment]
Getting:

```
The resource from “http://127.0.0.1:8081/main.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
```

In browser console when checking out the examples in test server, not sure if new

**Comment by Jyrno42** [comment]
> Getting:
> 
> ```
> The resource from “http://127.0.0.1:8081/main.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
> Warning: Expected server HTML to contain a matching <div> in <div>.
> ```
> 
> In browser console when checking out the examples in test server, not sure if new

Thats actually OK, it is for the case where docs have been deployed to github pages where the css file actually exists.

**Comment by simonschmidt** [comment]
Also in the example server, when closing a modal the page scrolls to top

**Comment by Jyrno42** [comment]
Can you try again now? I replaced the `position:fixed` and `overflow:hidden` with `body-scroll-lock` package. This package also deals with applying padding compensation for scrollbar of main page when needed. Since I am currently on a Mac can you please also verify that the scroll bar width is compensated for.

**Pull request merged**